### PR TITLE
feat: add workflows similar to amp-embedded-infra-lib, restructure components

### PR DESCRIPTION
**Pull request by richardapeters**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -89,11 +93,42 @@ jobs:
...
         if: always()
         with:
           files: test-report.xml
+
+  host_build_test:
+    name: Host Build & Test
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [macos-latest, windows-latest, windows-2019]
```
Comment (richardapeters) [inline .github/workflows/ci.yml:113]:
```suggestion
        os: [macos-latest, windows-latest]
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -38,9 +38,11 @@ jobs:
...
           key: ${{ github.job }}
           max-size: 2G
-      - run: |
-          cmake --preset=Windows
-          cmake --build --preset=Windows-Release
+      - uses: lukka/run-cmake@2ce8982be71b8e9a3c4d5e432135035afd1e76a7 # v10.7
+        with:
+          configurePreset: "Windows"
+          buildPreset: "Windows-Release"
+          configurePresetAdditionalArgs: "['-DCMAKE_C_COMPILER_LAUNCHER=ccache', '-DCMAKE_CXX_COMPILER_LAUNCHER=ccache']"
```
Comment (daantimmer) [inline .github/workflows/ci.yml:45]:
```suggestion
```

ccache is already configured in the 'defaults' preset

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -57,6 +59,13 @@ jobs:
           configurePreset: "Host"
           buildPreset: "Host-Release"
           testPreset: "Host-Release"
+          configurePresetAdditionalArgs: "['-DCMAKE_C_COMPILER_LAUNCHER=ccache', '-DCMAKE_CXX_COMPILER_LAUNCHER=ccache']"
```
Comment (daantimmer) [inline .github/workflows/ci.yml:62]:
```suggestion
```

ccache is already configured in the 'defaults' preset

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -73,6 +82,13 @@ jobs:
...
```
Comment (daantimmer) [inline .github/workflows/ci.yml:85]:
```suggestion
```

ccache is already configured in the 'defaults' preset

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -89,11 +105,42 @@ jobs:
         with:
           key: ${{ github.job }}
           max-size: 2G
+      - uses: lukka/run-cmake@af1be47fd7c933593f687731bc6fdbee024d3ff4 # v10.8
+        with:
+          configurePreset: "Host"
+          buildPreset: "Host-Debug"
+          configurePresetAdditionalArgs: "['-DCMAKE_C_COMPILER_LAUNCHER=ccache', '-DCMAKE_CXX_COMPILER_LAUNCHER=ccache']"
```
Comment (daantimmer) [inline .github/workflows/ci.yml:112]:
```suggestion
```

ccache is already configured in the 'defaults' preset

**Review by daantimmer — APPROVED** [review]

**Pull request merged**